evaluator/evaluator.py

```diff
@@ -244,22 +248,22 @@ def evaluate_vmaas(self, system_platform, cur):
                 unprocessed_cves.discard(cve)
                 if system_cves_map[cve]['when_mitigated']:
                     # it was mitigated, now its not
-                    unmitigated_cves.add(cve)
+                    unmitigated_cves_ids.add(self._cve_id_map[cve])
             else:
                 new_cves.add(cve)
 
         for cve in unprocessed_cves:
-            mitigated_cves.add(cve)
+            mitigated_cves_ids.add(self._cve_id_map[cve])
 
-        self._store_new_cves(system_id, inventory_id, new_cves, rh_account, cur)
-        self._update_mitigated_cves(system_id, [self._cve_id_map[cve] for cve in mitigated_cves], cur)
-        self._update_unmitigated_cves(system_id, [self._cve_id_map[cve] for cve in unmitigated_cves], cur)
+        new_cves_ids = self._store_new_cves(system_id, inventory_id, new_cves, rh_account, cur)
+        self._update_mitigated_cves(system_id, mitigated_cves_ids, cur)
+        self._update_unmitigated_cves(system_id, unmitigated_cves_ids, cur)
         self._update_system(system_id, len(reported_cves), cur)
         # don't update cve cache for opted out system
         if not opt_out:
             # pylint: disable=too-many-function-args
-            self._update_cve_affected_systems_cache(new_cves, unmitigated_cves,
-                                                    mitigated_cves, rh_account, cur)
+            self._update_cve_affected_systems_cache(new_cves_ids, unmitigated_cves_ids,
```

Comment:
this doesn't work correctly - it only updates cache when cve is inserted while processing system but it usually isn't
reproducer:
1. empty DB
2. evaluate first system => cache OK
3. evaluate second system with same packageset => cache not updated

Reply:
fixed